backport_packages/setup_backport_packages.py

```diff
@@ -422,13 +371,376 @@ def usage():
+    package_path = os.path.join(PROVIDERS_PATH, *package_id.split("."))
+    class_summary = get_package_class_summary(full_package_name)
+    past_releases = get_all_releases(package_path=package_path)
+    last_release = check_if_release_version_ok(past_releases, current_release_version)
+    cross_providers_dependencies = get_cross_provider_dependent_packages(package_id=package_id)
+    previous_release = get_previous_release(last_release=last_release, past_releases=past_releases,
+                                            current_release_version=current_release_version)
+    git_cmd = get_git_command(previous_release)
+    changes = subprocess.check_output(git_cmd, cwd=package_path, universal_newlines=True)
+    if changes == "":
+        print(f"The code has not changed since last release {last_release}. Skipping generating README.")
+        return
+    changes_table = convert_git_changes_to_table(
+        changes,
+        commit_url_prefix="https://github.com/apache/airflow/commit/")
+    pip_requirements_table = convert_pip_requirements_to_table(PROVIDERS_REQUIREMENTS[package_id])
+    cross_providers_dependencies_table = \
+        convert_cross_package_dependencies_to_table(
+            cross_providers_dependencies,
+            url_prefix="https://github.com/apache/airflow/tree/master/airflow/providers/")
+    context: Dict[str, Any] = {
+        "PACKAGE_ID": package_id,
+        "PACKAGE_PIP_NAME": f"apache-airflow-backport-providers-{package_id.replace('.', '-')}",
+        "FULL_PACKAGE_NAME": full_package_name,
+        "RELEASE": current_release_version,
+        "CURRENT_CHANGES_TABLE": changes_table,
+        "CROSS_PROVIDERS_DEPENDENCIES": cross_providers_dependencies,
+        "CROSS_PROVIDERS_DEPENDENCIES_TABLE": cross_providers_dependencies_table,
+        "PIP_REQUIREMENTS": PROVIDERS_REQUIREMENTS[package_id],
+        "PIP_REQUIREMENTS_TABLE": pip_requirements_table
+    }
+    current_changes = prepare_readme_from_template(template_name="PROVIDERS_CHANGES", context=context)
+    store_current_changes(package_path=package_path, current_release_version=current_release_version,
+                          current_changes=current_changes)
+    context.update(class_summary)
+    all_releases = get_all_releases(package_path)
+    context["RELEASES"] = all_releases
+    readme = LICENCE
+    readme += prepare_readme_from_template(template_name="PROVIDERS_README", context=context)
+    readme += prepare_readme_from_template(template_name="PROVIDERS_CLASSES", context=context)
+    for a_release in all_releases:
+        readme += a_release.content
+    readme_file_path = os.path.join(package_path, "README.md")
+    with open(readme_file_path, "wt") as readme_file:
+        readme_file.write(readme)
+    print(f"Generated {readme_file_path} file for the {package_id} provider")
+
+
+def import_all_providers_classes():
+    for loader, module_name, is_pkg in pkgutil.walk_packages([SOURCE_DIR_PATH]):
```

Comment:
Why do we need it?

Reply:
We need it because we want to find out the list of all operators/sensors/hooks/etc. We are using the usual mechanisms of python metaprogramming to discover all such operators and build documentation for all the operators automaticaly.

Reply:
I added more comments to the script now.